src/spikeinterface/sortingcomponents/matching/wobble.py

```diff
@@ -0,0 +1,929 @@
+    """Returns the length of the convolution of vectors with lengths x and y."""
+    return x + y - 1
+
+
+def convolve_templates(compressed_templates, params, template_meta, sparsity):
+    """Perform pairwise convolution on the compressed templates.
+
+    Parameters
+    ----------
+    compressed_templates : (ndarray, ndarray, ndarray, ndarray)
+        Templates compressed by singular value decomposition into temporal, singular, spatial, and
+        upsampled_temporal components.
+    params : WobbleParameters
+        Dataclass object for aggregating the parameters together.
+    template_meta : TemplateMetadata
+        Dataclass object for aggregating template metadata together.
+    sparsity : Sparsity
+        Dataclass object for aggregating channel sparsity variables together.
+
+    Returns
+    -------
+    pairwise_convolution : list[ndarray]
+        For each jittered template, pairwise_convolution of that template with each other overlapping template.
+    """
+    temporal, singular, spatial, temporal_jittered = compressed_templates
+    conv_res_len = get_convolution_len(template_meta.num_samples, template_meta.num_samples)
+    pairwise_convolution = []
+    for jittered_id in template_meta.jittered_indices:
+        num_overlap = np.sum(sparsity.unit_overlap[jittered_id, :])
+        template_id = jittered_id // params.jitter_factor
+        pconv = np.zeros([num_overlap, conv_res_len], dtype=np.float32)
+
+        # Reconstruct unit template from SVD Matrices
+        temporal_jittered_scaled = temporal_jittered[jittered_id] * singular[template_id][np.newaxis, :]
+        template_reconstructed = np.matmul(temporal_jittered_scaled, spatial[template_id, :, :])
+        template_reconstructed = np.flipud(template_reconstructed)
+
+        units_are_overlapping = sparsity.unit_overlap[jittered_id, :]
+        overlapping_units = np.where(units_are_overlapping)[0]
+        for j, jittered_id2 in enumerate(overlapping_units):
+            temporal_overlapped = temporal[jittered_id2]
+            singular_overlapped = singular[jittered_id2]
+            spatial_overlapped = spatial[jittered_id2]
+            visible_overlapped_channels = sparsity.visible_channels[jittered_id2, :]
+            visible_template = template_reconstructed[:, visible_overlapped_channels]
+            spatial_filters = spatial_overlapped[:params.conv_approx_rank, visible_overlapped_channels].T
+            spatially_filtered_template = np.matmul(visible_template, spatial_filters)
+            scaled_filtered_template = spatially_filtered_template * singular_overlapped
+            for i in range(params.conv_approx_rank):
+                pconv[j, :] += np.convolve(scaled_filtered_template[:, i], temporal_overlapped[:, i],
+                                                   'full')
+        pairwise_convolution.append(pconv)
+    return pairwise_convolution
+
+
+def compute_objective(traces, template_data, conv_approx_rank):
```

Comment:
This function is also doing two things. Computing and normalizing. I think they should be separated.